lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -488,8 +488,10 @@ public function diffColumn(Column $column1, Column $column2)
         $platformOptions1 = $column1->getPlatformOptions();
         $platformOptions2 = $column2->getPlatformOptions();
 
-        foreach (array_keys(array_intersect_key($platformOptions1, $platformOptions2)) as $key) {
-            if ($properties1[$key] !== $properties2[$key]) {
+        foreach (array_merge(array_keys($platformOptions1), array_keys($platformOptions2)) as $key) {
```

Comment:
Looking at this again, it seems that it should be doing already what I was suggesting, even though a lit bit more complicated. So you might just add some more tests to reveal any issues.